### DISCUSSION TITLE
There should be a way to call a primitive type's function by string.

### DISCUSSION THREAD

**Discussion by caimantilla** [godotengine/godot-proposals#6878] [Scripting]
It was mildly frustrating to find out that `Callable` only works with objects.

My understanding is that adding this functionality to non-Object types would add a lot of overhead, so I'm not asking for it to support those types. I sure don't want to grind performance to a halt just so that I can store a reference to `Vector3i.clamped`.
However, wouldn't it make sense if maybe there were some wrapper class that does this? Like `ArrayCallable`, or `Vector3Callable`, which would store a copy or (preferably) reference of that type and allow calling its functions that way. This could be done in GDScript, but it's... kind of a pain to solve what should be a simple problem. Execution speed would also suffer, since IIRC `match` statements in GDScript are syntactic sugar and every condition still needs to be checked.

A real-world example of this can be seen [here](https://github.com/nathanhoad/godot_dialogue_manager/blob/main/addons/dialogue_manager/dialogue_manager.gd?plain=1#L499). I was trying to call `Array.pop_front` in a while loop in my dialogue script, only to find out that every single `Array` function has to be hard-coded, and `pop_front` wasn't implemented. So, while not entirely necessary, a native class for this type of case would be incredibly helpful, although a bit of extra maintenance would be required if it can't be auto-generated.

Note 1: Maybe accepting a `Variant` and being called `PrimitiveCallable` would be ideal?
Note 2: It also might only make sense to support this for reference types (`Array`, `Dictionary`, packed arrays...)

**Comment by theraot** [comment]
You can already write your own method on your script and make a `Callable` out of it, or directly use a "lambda", which I believe is a good approach since you want a reference to the value and not a copy. The hassle, of course, would be making one for each method you want, but I think it will do for your situation at hand, as apparently you only need `pop_front`.

**Comment by dalexeev** [comment]
This is implemented for future 4.3:

* godotengine/godot#82264
* godotengine/godot#88948
